src/Compilers/Core/Portable/StrongName/CryptoBlobParser.cs

```diff
@@ -91,53 +91,45 @@ internal static bool IsValidPublicKey(ImmutableArray<byte> blob)
                 return false;
             }
 
-            return IsValidPublicKeyUnsafe(blob);
-        }
+            BinaryReader blobReader = new BinaryReader(new MemoryStream(blob.DangerousGetUnderlyingArray()));
```

Comment:
This is a read only operation and neither of the types have a finalizer hence I don't think we strictly need them. However we should add them because it's best practice here (and saves me trouble of digging around to remember if these types have finalizers or not).